codeworld-api/src/CodeWorld/CanvasM.hs

```diff
@@ -85,6 +85,56 @@ saveRestore m = do
 
 #ifdef ghcjs_HOST_OS
 
+newtype StubCanvasM a = StubCanvasM
+    { unStubCanvasM :: IO a
+    } deriving (Functor)
+
+
+instance Applicative StubCanvasM where
+    pure x = StubCanvasM (return x)
+    f <*> x = StubCanvasM (unStubCanvasM f <*> unStubCanvasM x)
+
+instance Monad StubCanvasM where
+    return = pure
+    m >>= f = StubCanvasM (unStubCanvasM m >>= unStubCanvasM . f)
+
+instance MonadIO StubCanvasM where
+    liftIO = StubCanvasM
+
+instance MonadCanvas StubCanvasM where
+    type Image StubCanvasM = ()
+    save = StubCanvasM $ return ()
```

Comment:
I suppose since you have a Monad instance already, there's no need to explicitly wrap in StubCanvasM.  You could just write `save = return ()`.  Same for others.